src/server/scripts/Spells/spell_item.cpp

```diff
@@ -821,6 +822,60 @@ class spell_item_deviate_fish : public SpellScript
     }
 };
 
+class PartyTimeEmoteEvent : public BasicEvent
+{
+public:
+    PartyTimeEmoteEvent(Player* player) : _player(player) { }
+
+    bool Execute(uint64 /*time*/, uint32 /*diff*/) override
+    {
+        if (_player->isMoving())
+            _player->HandleEmoteCommand(RAND(EMOTE_ONESHOT_APPLAUD, EMOTE_ONESHOT_LAUGH, EMOTE_ONESHOT_CHEER, EMOTE_ONESHOT_CHICKEN));
+        else
+            _player->HandleEmoteCommand(RAND(EMOTE_ONESHOT_APPLAUD, EMOTE_ONESHOT_DANCESPECIAL, EMOTE_ONESHOT_LAUGH, EMOTE_ONESHOT_CHEER, EMOTE_ONESHOT_CHICKEN));
+
+        _player->m_Events.AddEventAtOffset(this, RAND(5s, 10s, 15s));
+
+        return false; // do not delete re-added event in EventProcessor::Update
+    }
+
+private:
+    Player* _player;
+};
+
+class spell_item_party_time : public AuraScript
+{
+    PrepareAuraScript(spell_item_party_time);
+
+    void HandleEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Player* player = GetOwner()->ToPlayer();
+        if (!player)
+            return;
+
+        _event = new PartyTimeEmoteEvent(player);
+        player->m_Events.AddEventAtOffset(_event, RAND(5s, 10s, 15s));
+    }
+
+    void HandleEffectRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (!_event)
```

Comment:
Are auras removed before events when logging out so that _event will never point to deleted memory ?

Reply:
i wonder... in logout all m_Events are not deleted anyway?

Reply:
@jackpoz good point. I can override `BasicEvent::IsDeletable()`, but your idea with aura check inside event looks better.
@Keader they are deleted. This is the problem - my pointer will point to deleted memory.